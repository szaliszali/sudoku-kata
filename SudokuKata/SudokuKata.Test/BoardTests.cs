using System;
using ApprovalTests;
using NUnit.Framework;

namespace SudokuKata.Test;

[TestFixture]
internal class BoardTests
{
    [Test]
    public void EmptyBoardToString() => Approvals.Verify(new CharArrayBoard());

    [Test]
    public void EmptyBoardCode() => Approvals.Verify(new CharArrayBoard().Code);

    [Test]
    public void RandomBoard()
    {
        var rng = new Random(1);
        var sut = new RandomBoard(rng);
        Approvals.Verify((sut, sut.Code));
    }

    [Test]
    public void SetCellToDigit()
    {
        var sut = new CharArrayBoard();
        sut.Set(1, 2, 3);
        Approvals.Verify(sut);
    }

    [Test]
    public void ClearCell()
    {
        var sut = new CharArrayBoard();
        sut.Set(1, 2, 3);
        sut.Set(1, 2, 0);
        Approvals.Verify(sut);
    }

    [Test]
    public void ConstructFromStateArray()
    {
        var state = new[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
        var sut = new CharArrayBoard(state);
        Approvals.Verify((sut, sut.Code));
    }

    [Test]
    public void ConstructFromStateArray_StateIsPreserved()
    {
        var state = new[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
        var sut = new CharArrayBoard(state);
        Assert.That(sut.State, Is.EqualTo(state));
    }
}
