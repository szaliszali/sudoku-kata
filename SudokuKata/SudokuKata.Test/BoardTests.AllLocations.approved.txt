[0] = CellLocation { Row = 0, Column = 0 }
[1] = CellLocation { Row = 0, Column = 1 }
[2] = CellLocation { Row = 0, Column = 2 }
[3] = CellLocation { Row = 0, Column = 3 }
[4] = CellLocation { Row = 0, Column = 4 }
[5] = CellLocation { Row = 0, Column = 5 }
[6] = CellLocation { Row = 0, Column = 6 }
[7] = CellLocation { Row = 0, Column = 7 }
[8] = CellLocation { Row = 0, Column = 8 }
[9] = CellLocation { Row = 1, Column = 0 }
[10] = CellLocation { Row = 1, Column = 1 }
[11] = CellLocation { Row = 1, Column = 2 }
[12] = CellLocation { Row = 1, Column = 3 }
[13] = CellLocation { Row = 1, Column = 4 }
[14] = CellLocation { Row = 1, Column = 5 }
[15] = CellLocation { Row = 1, Column = 6 }
[16] = CellLocation { Row = 1, Column = 7 }
[17] = CellLocation { Row = 1, Column = 8 }
[18] = CellLocation { Row = 2, Column = 0 }
[19] = CellLocation { Row = 2, Column = 1 }
[20] = CellLocation { Row = 2, Column = 2 }
[21] = CellLocation { Row = 2, Column = 3 }
[22] = CellLocation { Row = 2, Column = 4 }
[23] = CellLocation { Row = 2, Column = 5 }
[24] = CellLocation { Row = 2, Column = 6 }
[25] = CellLocation { Row = 2, Column = 7 }
[26] = CellLocation { Row = 2, Column = 8 }
[27] = CellLocation { Row = 3, Column = 0 }
[28] = CellLocation { Row = 3, Column = 1 }
[29] = CellLocation { Row = 3, Column = 2 }
[30] = CellLocation { Row = 3, Column = 3 }
[31] = CellLocation { Row = 3, Column = 4 }
[32] = CellLocation { Row = 3, Column = 5 }
[33] = CellLocation { Row = 3, Column = 6 }
[34] = CellLocation { Row = 3, Column = 7 }
[35] = CellLocation { Row = 3, Column = 8 }
[36] = CellLocation { Row = 4, Column = 0 }
[37] = CellLocation { Row = 4, Column = 1 }
[38] = CellLocation { Row = 4, Column = 2 }
[39] = CellLocation { Row = 4, Column = 3 }
[40] = CellLocation { Row = 4, Column = 4 }
[41] = CellLocation { Row = 4, Column = 5 }
[42] = CellLocation { Row = 4, Column = 6 }
[43] = CellLocation { Row = 4, Column = 7 }
[44] = CellLocation { Row = 4, Column = 8 }
[45] = CellLocation { Row = 5, Column = 0 }
[46] = CellLocation { Row = 5, Column = 1 }
[47] = CellLocation { Row = 5, Column = 2 }
[48] = CellLocation { Row = 5, Column = 3 }
[49] = CellLocation { Row = 5, Column = 4 }
[50] = CellLocation { Row = 5, Column = 5 }
[51] = CellLocation { Row = 5, Column = 6 }
[52] = CellLocation { Row = 5, Column = 7 }
[53] = CellLocation { Row = 5, Column = 8 }
[54] = CellLocation { Row = 6, Column = 0 }
[55] = CellLocation { Row = 6, Column = 1 }
[56] = CellLocation { Row = 6, Column = 2 }
[57] = CellLocation { Row = 6, Column = 3 }
[58] = CellLocation { Row = 6, Column = 4 }
[59] = CellLocation { Row = 6, Column = 5 }
[60] = CellLocation { Row = 6, Column = 6 }
[61] = CellLocation { Row = 6, Column = 7 }
[62] = CellLocation { Row = 6, Column = 8 }
[63] = CellLocation { Row = 7, Column = 0 }
[64] = CellLocation { Row = 7, Column = 1 }
[65] = CellLocation { Row = 7, Column = 2 }
[66] = CellLocation { Row = 7, Column = 3 }
[67] = CellLocation { Row = 7, Column = 4 }
[68] = CellLocation { Row = 7, Column = 5 }
[69] = CellLocation { Row = 7, Column = 6 }
[70] = CellLocation { Row = 7, Column = 7 }
[71] = CellLocation { Row = 7, Column = 8 }
[72] = CellLocation { Row = 8, Column = 0 }
[73] = CellLocation { Row = 8, Column = 1 }
[74] = CellLocation { Row = 8, Column = 2 }
[75] = CellLocation { Row = 8, Column = 3 }
[76] = CellLocation { Row = 8, Column = 4 }
[77] = CellLocation { Row = 8, Column = 5 }
[78] = CellLocation { Row = 8, Column = 6 }
[79] = CellLocation { Row = 8, Column = 7 }
[80] = CellLocation { Row = 8, Column = 8 }
